### PR TITLE
Added support for VirtualNetworkUsages API to Virtual Networks

### PR DESCRIPTION
**Pull request by anton-evseev**
## Description
Added support for VirtualNetworkUsages API to Virtual Networks
Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/1168

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

File: src/SDKs/Network/Network.Tests/Tests/VirtualNetworkTests.cs [hunk 1]
```diff
@@ -372,5 +372,105 @@ public void VirtualNetworkPeeringTest()
                 networkManagementClient.VirtualNetworks.Delete(resourceGroupName, vnet2Name);
             }
         }
+
+        [Fact]
```
Comment (shahabhijeet) [inline src/SDKs/Network/Network.Tests/Tests/VirtualNetworkTests.cs]: @number213 all new tests should be using latest Resource Manager nuget package.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Comment by number213** [comment]
@shahabhijeet re-recorded new test with the latest Resource Manager

File: src/SDKs/Network/Network.Tests/Tests/VirtualNetworkTests.cs [hunk 2]
```diff
@@ -372,5 +372,105 @@ public void VirtualNetworkPeeringTest()
...
+        public void VirtualNetworkUsageTest()
+        {
+            var handler1 = new RecordedDelegatingHandler { StatusCodeToReturn = HttpStatusCode.OK };
+            var handler2 = new RecordedDelegatingHandler { StatusCodeToReturn = HttpStatusCode.OK };
+
+            using (MockContext context = MockContext.Start(this.GetType().FullName))
+            {
+                var resourcesClient = ResourcesManagementTestUtilities.GetResourceManagementClientWithHandler(context, handler1);
```
Comment (shahabhijeet) [inline src/SDKs/Network/Network.Tests/Tests/VirtualNetworkTests.cs]: @number213 I don't see any changes to code that creates the actual Resource Manager client.
Could you point me to that change?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Comment by number213** [comment]
@shahabhijeet sorry for taking so long, finally got what you meant. Updated Virtual Network tests to use latest Resource Manager instead of depreciated one.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**